src/basics.md

```diff
@@ -277,6 +278,54 @@ fn main() {
 }
 ```
 
+[ex-rand-passwd]: #ex-rand-passwd
+<a name="ex-rand-passwd"></a>
+## Create random passwords from a set of allowed characters
+
+[![rand-badge]][rand]
+
+Randomly generates a string of given length ASCII characters in the range `A-Z, a-z, 0-9`, with [`gen_ascii_chars`] or [`choose`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+# 
+# error_chain! {
+#     foreign_links {
+#         String(std::string::FromUtf8Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let s: String = thread_rng().gen_ascii_chars().take(30).collect();
+    println!("{}", s);
+
+    let alphanum_bytestring =
+        String::from_utf8((b'a'..b'z')
+        .chain(b'A'..b'Z')
+        .chain(b'0'..b'9')
+        //.chain([b'!', b'?', b'%'])
```

Comment:
It does not work because ranges give you `u8` while iterating over slice gives you `&u8` so the solution (if we wanted to go with this approach and we do not ;)) would be to clone the iterator contents.
        `.chain([b'!', b'?', b'%'].iter().cloned())` or even better  `.chain(b"!?%".iter().cloned())`.

On the other hand I would suggest to just go with bytestring literal
```
const CHARSET: &[u8] =  b"ABCDEFGHIJKLMNOPQRSTUVWXYZ\
              abcdefghijklmnopqrstuvwxyz\
              0123456789)(*&^%$#@!~";
```